pkg/kb_agent/httpserver/server.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package httpserver
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	fasthttprouter "github.com/fasthttp/router"
+	"github.com/valyala/fasthttp"
+)
+
+// Server is an interface for the kb-agent HTTP server.
+type Server interface {
+	io.Closer
+	Router() fasthttp.RequestHandler
+	StartNonBlocking() error
+}
+
+type server struct {
+	config    Config
+	endpoints []Endpoint
+	servers   []*fasthttp.Server
+}
+
+// NewServer returns a new HTTP server.
+func NewServer() Server {
+	return &server{
+		endpoints: Endpoints(),
+		config:    config,
+	}
+}
+
+// StartNonBlocking starts a new server in a goroutine.
+func (s *server) StartNonBlocking() error {
```

Comment:
How many concurrent requests (goroutines) that a server can handle?

Reply:
The code should add a configuration for explicitly limiting the number of concurrent requests to prevent overload.

Reply:
got, the default max concurrency is 256 * 1024 which is used by fasthttps. for kb-agent it is set to 1024 now.

Reply:
should be even smaller?

Reply:
yes, under normal circumstances, the number of concurrent currency connections does not exceed 10. so the default value can be set to 10?